src/microservices/Updating/Microservices.UpdateValues/Execution/UpdateValuesQueueConsumer.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Rdmp.Core.Repositories;
+using Smi.Common.Messages;
+using Smi.Common.Messages.Updating;
+using Smi.Common.Messaging;
+using Smi.Common.Options;
+
+namespace Microservices.UpdateValues.Execution
+{
+    public class UpdateValuesQueueConsumer : Consumer<UpdateValuesMessage>
+    {
+        private Updater _updater;
+
+        public UpdateValuesQueueConsumer(UpdateValuesOptions opts,ICatalogueRepository repo)
+        {
+            _updater = new Updater(repo);
+            _updater.UpdateTimeout = opts.UpdateTimeout;
+            _updater.TableInfosToUpdate = opts.TableInfosToUpdate;
+        }
+        protected override void ProcessMessageImpl(IMessageHeader header, UpdateValuesMessage message, ulong tag)
+        {
+            _updater.HandleUpdate(message);
```

Comment:
Are there any "known" exceptions we should catch here and Nack? Currently any exception will cause the service to crash.

E.g. I think in other services we Nack (throw away) the message if it is invalid in some way, i.e. permanently broken, but allow the service to continue running.

Reply:
Hmn... good question.  Things like corrupt messages (ones that don't pass validation) would be candidates.  But at least to start with I'd rather those hard crash since we really shouldn't be generating those.  We are likely to get some issues with performance that will require indexes to be built etc so I see spinning this up as an iterative process.  We might want to add automatic temporary index creation and/or updating multiple tables/servers at once.  Theres a lot of ways we could improve this service so starting out basic might be best.